bonfire/Modules/Tools/Controllers/LogsController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Bonfire\Tools\Controllers;
+
+use App\Controllers\AdminController;
+use CodeIgniter\CodeIgniter;
+use Bonfire\Modules\Tools\Libraries\Logs;
+
+class LogsController extends AdminController
+{
+    protected $theme = 'Admin';
+
+    protected $viewPrefix = 'Bonfire\Modules\Tools\Views\\';
+
+    protected $logsPath = WRITEPATH . 'logs/';
+
+    public function __construct()
+    {
+        $this->logsLimit = service('settings')->get('Logs.paginationLimit');
+
+        $this->logsHandler = new Logs();
+    }
+
+    /**
+     * Displays all logs.
+     *
+     * @return string
+     */
+    public function index()
+    {
+        // Load the Log Files.
+        $logs = get_filenames($this->logsPath);
+
+        unset($logs[0]);
+
+        $result = $this->logsHandler->paginateLogs($logs,$this->logsLimit);
+
+        return $this->render($this->viewPrefix .'logs', [
+            'logs' => $result['logs'],
+            'pager' => $result['pager'],
+        ]);
+    }
+
+    /**
+     * Show the contents of a single log file.
+     *
+     * @param string $file The full name of the file to view (including extension).
+     *
+     * @return void
+     */
+    public function view(string $file = '')
+    {
+        if (empty($file) or !file_exists($this->logsPath.$file)) {
+            return redirect()->to(ADMIN_AREA . 'tools/logs')->with('danger', lang('Logs.empty'));
+        }
+
+        $logs = $this->logsHandler->processFileLogs($this->logsPath.$file);
+
+        $result = $this->logsHandler->paginateLogs($logs,$this->logsLimit);
+
+        return $this->render($this->viewPrefix .'view', [
+            'log_file' => $file,
+            'canDelete' => 1,
+            'log_content' => $result['logs'],
+            'pager' => $result['pager'],
+            'log_file_pretty' => date('F j, Y', strtotime(str_replace('.log', '', str_replace('log-', '', $file)))),
+        ]);
+
+    }
+
+    /**
+     * Delete the specified log file or all.
+     *
+     * @return \CodeIgniter\HTTP\RedirectResponse
+     */
+    public function delete(){
+
+      if (isset($_POST['delete'])) {
+
+          $checked = $_POST['checked'];
+          $numChecked = count($checked);
+
+          if (is_array($checked) && $numChecked) {
+
+              foreach ($checked as $file) {
+                  @unlink($this->logsPath . $file);
```

Comment:
Since we're getting  a filename here through POST, we need to be secure and sanitize the filename. The security helper has a function for that.